forage-android/src/main/java/com/joinforage/forage/android/collect/VgsPinSubmitter.kt

```diff
@@ -0,0 +1,159 @@
+package com.joinforage.forage.android.collect
+
+import android.content.Context
+import com.joinforage.forage.android.VaultType
+import com.joinforage.forage.android.core.StopgapGlobalState
+import com.joinforage.forage.android.model.EncryptionKeys
+import com.joinforage.forage.android.model.PaymentMethod
+import com.joinforage.forage.android.network.ForageConstants
+import com.joinforage.forage.android.network.model.ForageApiError
+import com.joinforage.forage.android.network.model.ForageApiResponse
+import com.joinforage.forage.android.network.model.ForageError
+import com.joinforage.forage.android.network.model.UnknownErrorApiResponse
+import com.joinforage.forage.android.pos.PosVaultProxyRequest
+import com.joinforage.forage.android.ui.ForagePINEditText
+import com.verygoodsecurity.vgscollect.VGSCollectLogger
+import com.verygoodsecurity.vgscollect.core.HTTPMethod
+import com.verygoodsecurity.vgscollect.core.VGSCollect
+import com.verygoodsecurity.vgscollect.core.VgsCollectResponseListener
+import com.verygoodsecurity.vgscollect.core.model.network.VGSRequest
+import com.verygoodsecurity.vgscollect.core.model.network.VGSResponse
+import org.json.JSONException
+import kotlin.coroutines.suspendCoroutine
+
+internal class VgsPinSubmitter(
+    context: Context,
+    foragePinEditText: ForagePINEditText
+) : AbstractVaultSubmitter<VGSResponse?>(
+    context = context,
+    foragePinEditText = foragePinEditText,
+    vaultType = VaultType.VGS_VAULT_TYPE
+) {
+    override suspend fun submitProxyRequest(
+        vaultProxyRequest: VaultProxyRequest
+    ): ForageApiResponse<String> = suspendCoroutine { continuation ->
+        val vgsCollect = buildVGSCollect(context)
+        vgsCollect.bindView(foragePinEditText.getTextInputEditText())
+        val vgsInputField = foragePinEditText.getTextInputEditText()
+
+        vgsCollect.addOnResponseListeners(object : VgsCollectResponseListener {
+            override fun onResponse(response: VGSResponse?) {
+                // VGS wants us to call onDestroy, unlike Basis Theory
+                vgsCollect.onDestroy()
+                vgsInputField.setText("")
```

Comment:
why do `ForagePINEditText.clearText()`([see code](https://github.com/teamforage/forage-android-sdk/blob/79dc8fd508d60fd5fbf5cc7b07a2a122dcc904d3/forage-android/src/main/java/com/joinforage/forage/android/ui/VGSVaultWrapper.kt#L120)) and the clearText inside here have different behaviors?

Reply:
You can call `foragePinEditText.clearText()` or `vgsInputField.setText("")`. Both would work here.

Reply:
do we still have to call `vgsCollect.onDestroy()` @dleis612 ?

Reply:
Uhhhhh I'm not entirely sure I know what that call does actually.

Reply:
According to the VGS method comment:


```kotlin
// Clear all information collected before by VGSCollect. Preferably call it inside onDestroy system's callback.
 fun onDestroy() {
        client.cancelAll()
        responseListeners.clear()
        storage.clear()
    }
```

Reply:
Seems like we should continue calling it.